vispy/scene/widgets/grid.py

```diff
@@ -56,7 +57,9 @@ def add_widget(self, widget=None, row=None, col=None, row_span=1,
 
         _row = self._cells.setdefault(row, {})
         _row[col] = widget
-        self._grid_widgets[widget] = row, col, row_span, col_span
+        self._grid_widgets[self._n_added] = (row, col, row_span, col_span,
```

Comment:
yeah, I know -- it's more for future compatibility in case we want to be able to delete axes and such. This way we can `.pop` without needing to update index numbers or anything.